### ISSUE TITLE
[MNG-2083] Path to missing dependency is not shown nor url to download 

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#4142]
**[Carlos Sanchez Gonzalez](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=carlos)** opened **[MNG-2083](https://issues.apache.org/jira/browse/MNG-2083?redirect=false)** and commented

Seems a regression
it's only shown with -X

[DEBUG] Unable to download the artifact from any repository

Try downloading the file manually from
http://java.sun.com/products/javabeans/glasgow/jaf.html
and install it using the command:
mvn install:install-file -DgroupId=javax.activation -DartifactId=activation -Dversion=1.0.2 -Dpackaging=jar -Dfile=/path/to/file
Path to dependency:
1) test:test:jar:1.0-SNAPSHOT
2) javax.activation:activation:jar:1.0.2

javax.activation:activation:jar:1.0.2

---

**Affects:** 2.0.3

**Attachments:**
- [pom.xml](https://issues.apache.org/jira/secure/attachment/12711897/pom.xml) (_513 bytes_)
- [pom.xml](https://issues.apache.org/jira/secure/attachment/12711557/pom.xml) (_359 bytes_)

**Issue Links:**
- [MNG-1791](https://issues.apache.org/jira/browse/MNG-1791) Instructions to download missing jar have dissapeared
 (_**"fixes"**_)
- [MNG-338](https://issues.apache.org/jira/browse/MNG-338) provide some hand holding for dependencies on non-distributable jars

**Comment by jira-importer** [comment]
**[John Dennis Casey](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jdcasey)** commented

ready for merge from trunk.

**Comment by jira-importer** [comment]
**[John Dennis Casey](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jdcasey)** commented

fixed and merged.

**Comment by jira-importer** [comment]
**[Carlos Sanchez Gonzalez](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=carlos)** commented

There's a problem when more than one dependecy is missing, some info (install instructions, download url) for the first one is ommitted

2 required artifacts missing:

1) javax.activation:activation:jar:1.0.2
   Path to dependency:

   1) test:test:jar:1.0-SNAPSHOT
   2) javax.activation:activation:jar:1.0.2
2) javax.mail:mail:jar:1.3.2

Try downloading the file manually from
http://java.sun.com/products/javamail/downloads/index.html

Then, install it using the command:
mvn install:install-file -DgroupId=javax.mail -DartifactId=mail \
-Dversion=1.3.2 -Dpackaging=jar -Dfile=/path/to/file

Path to dependency:
1) test:test:jar:1.0-SNAPSHOT
2) javax.mail:mail:jar:1.3.2

for the artifact:
test:test:jar:1.0-SNAPSHOT

from the specified remote repositories:
central (http://test.maven.codehaus.org/maven2)

**Comment by jira-importer** [comment]
**[Carlos Sanchez Gonzalez](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=carlos)** commented

The pom with two missing dependencies

**Comment by jira-importer** [comment]
**[John Dennis Casey](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jdcasey)** commented

there was a problem with projects that were already cached...the pomArtifact doesn't get populated with downloadUrl, because the project is already in the cache.

I corrected this to set the downloadUrl on both the main artifact and the pomArtifact, for consistency. Then, I added a little bit more formatting to try to make multi-artifact output more legible.
